lti_consumer/plugin/views.py

```diff
@@ -415,3 +418,130 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LTINRPSMembersipPagination
+
+    def parse_role_filter(self):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        role_filter_param = self.request.query_params.get('role', None)
+
+        matched_role = None
+        if role_filter_param:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if role_filter_param in context_roles:
+                    matched_role = edx_role
+
+        return matched_role
+
+    def get_memberships(self, kwarg_overrides=None):
+        """
+        Helper method to create queryset.
+        """
+        # get course key
+        course_key = self.request.lti_configuration.location.course_key
+
+        kwargs = {
+            'access_roles': [],
+            'include_students': True,
+        }
+
+        if kwarg_overrides and isinstance(kwarg_overrides, dict):
+            kwargs.update(kwarg_overrides)
+
+        return partial(compat.get_course_members, course_key, **kwargs)
+
+    def get_queryset(self):
+        """
+        Get User queryset related to current course. Also filter via role if provided.
+        """
+        kwargs = {}
```

Comment:
Avoid using a generic parameter dict, create separate variables for each required parameter.
```
access_roles = []
include_students = False
limit = self.request.query_params.get('limit', None)
page = self.request.query_params.get('p', None)
course_key = ...

...

return attach_external_id_information(
    compat.get_course_members(...)
)
```

Reply:
Now using explicit variables rather than generic kwarg dict.